### PR TITLE
Remove deprecated docker infrastructure

### PR DESCRIPTION
**Pull request by skabashnyuk**
### What does this PR do?
Remove deprecated docker infrastructure

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/14356

#### Release Notes
n/a

#### Docs PR
n/a

### CODE REVIEW

**Comment by skabashnyuk** [comment]
@eclipse/eclipse-che-qa can you take a look at tests results?

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/core/provider/CheTestWorkspaceAgentApiEndpointUrlProvider.java [hunk 1]
```diff
@@ -48,4 +48,14 @@ public URL get(String workspaceId) throws Exception {
...
   }
+
+  /**
+   * Checks whether provided {@link Machine} contains wsagent server.
+   *
+   * @param machine machine to check
+   * @return true when wsagent server is found in provided machine, false otherwise
+   */
+  public static boolean containsWsAgentServer(Machine machine) {
+    return machine.getServers().keySet().contains(Constants.SERVER_WS_AGENT_HTTP_REFERENCE);
```
Comment (sleshchenko) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/core/provider/CheTestWorkspaceAgentApiEndpointUrlProvider.java:59]: Actually it's a good time to get rid of `Constants.SERVER_WS_AGENT_HTTP_REFERENCE` as well.

P.S. Along with other unused constants like `SERVER_WS_AGENT_WEBSOCKET_REFERENCE`, `SERVER_EXEC_AGENT_WEBSOCKET_REFERENCE`, `WS_AGENT_PORT`

**Review by sleshchenko — COMMENTED** [review]
Are there any plans to get rid of docker infra in selenium tests? Maybe in scope of another issue for QE team?

**Review by sleshchenko — COMMENTED** [review]
Consider removing `AbstractBootstrapper`:
```java
 * @deprecated should be removed along with the docker infrastructure
 */
@Deprecated
public abstract class AbstractBootstrapper {
```

**Review by sleshchenko — COMMENTED** [review]
To avoid confusing it would be better to update example for Recipe class https://github.com/eclipse/che/blob/master/core/che-core-api-model/src/main/java/org/eclipse/che/api/core/model/workspace/config/Recipe.java#L20

**Review by sleshchenko — APPROVED** [review]
Please take a look my inline comments
Other LGTM

**Comment by skabashnyuk** [comment]
@sleshchenko 
- AbstractBootstrapper - removed
- Recipe.java#L20 - cleaned up.
- Constants - propose to handle in separate pr since it requires additional cleaning in K8s/Os infrastructures. And I don't want to mix this change with cleaning since it would be hard to review.

**Comment by skabashnyuk** [comment]
crw-ci-test

**Review by amisevsk — COMMENTED** [review]
Looks like the following properties should also be removed:

```
che.dns.resolvers
che.workspace.hosts
che.docker.always_pull_image
che.docker.privileged
che.docker.pids_limit
che.docker.parent_cgroup
che.docker.cpuset_cpus
che.docker.volumes_projects_options
che.docker.volumes_agent_options
che.docker.ip
che.docker.ip.external
che.docker.cleanup_period_min
che.docker.api
che.docker.enable_container_stop_detector
che.docker.cpu_period
che.docker.cpu_quota
che.docker.network_driver
che.docker.securityopt
che.docker.tcp_connection_timeout_ms
che.docker.tcp_connection_read_timeout_ms
che.docker.swap
che.infra.docker.master_api_endpoint
che.infra.docker.master_websocket_endpoint
che.infra.docker.master_websocket_minor_endpoint
che.infra.docker.bootstrapper.timeout_min
che.infra.docker.bootstrapper.installer_timeout_sec
che.infra.docker.bootstrapper.server_check_period_sec
che.infra.docker.max_pull_threads
che.infra.docker.build_timeout_sec
che.single.port
che.infra.docker.url_rewriter
che.workspace.migrate_workspace_projects_on_startup
che.singleport.wildcard_domain.host
```

**Review by amisevsk — APPROVED** [review]

**Pull request merged**